include/tcframe/spec/io/IOFormat.hpp

```diff
@@ -101,25 +117,40 @@ class IOFormatBuilder {
     void addLastSegment() {
         if (lastBuilder_ != nullptr) {
             if (currentFormat_ != nullptr) {
-                if (hasLinesSegmentWithoutSizeAsLastSegment()) {
-                    throw runtime_error("Lines segment without size can only be the last segment");
-                }
+                checkSegments();
                 currentFormat_->push_back(lastBuilder_->build());
             }
             lastBuilder_ = nullptr;
         }
     }
 
-    bool hasLinesSegmentWithoutSizeAsLastSegment() {
+    void checkSegments() {
         if (currentFormat_->empty()) {
-            return false;
+            return;
+        }
```

Comment:
``IOSegment* lastSegment`` is computed twice here.

The methods ``hasLinesSegmentWithoutSizeAsLastSegment()`` and ``hasRawLinesSegmentWithoutSizeAsLastSegment()`` are not modular, since it does not check whether the ``currentFormat_`` is empty and it requires the caller (in this case ``checkSegments()`` ) to check the emptiness first.

Probably we can compute the ``IOSegment* lastSegment`` in here, and then pass in the value in the ``hasLinesSegmentWithoutSizeAsLastSegment()`` and ``hasRawLinesSegmentWithoutSizeAsLastSegment()`` method. Then we need to rename the methods name though.